src/demangle.cpp

```diff
@@ -117,14 +60,19 @@ void Demangle::cutArguments(char* s) {
 }
 
 char* Demangle::demangle(const char* s, bool full_signature) {
-    // Check if the mangled symbol ends with a Rust hash "17h<hex>E"
-    const char* e = strrchr(s, 'E');
-    if (e != NULL && e - s > 22 && e[-19] == '1' && e[-18] == '7' && e[-17] == 'h') {
-        const char* h = e - 16;
-        while ((*h >= '0' && *h <= '9') || (*h >= 'a' && *h <= 'f')) h++;
-        if (h == e) {
-            return demangleRust(s + 3, e - 19);
-        }
+    // try to demangle as Rust
+    struct demangle demangle;
+    rust_demangle_demangle(s, &demangle);
```

Comment:
This runs some non-trivial pre-parsing. I wonder if this may have performance impact, given that the code is also executed for every C++ symbol.

Reply:
it should be fairly fast. most C++ symbols don't start with _ZN and a digit, and even if they do it just follows the list of lengths.

Reply:
> most C++ symbols don't start with _ZN and a digit

The majority of JVM symbols start with _ZN and a digit:
```
nm /usr/java/jdk-23.0.1/lib/server/libjvm.so | grep " _ZN[1-9]" | wc -l
34181
```

Reply:
Some of these symbols are demangled as Rust producing a different result, e.g.:
```
_ZN12_GLOBAL__N_113single_threadE
Rust: _GLOBAL__N_1::single_thread
CPP: (anonymous namespace)::single_thread
```